src/CGIHandler.cpp

```diff
@@ -103,26 +110,39 @@ void handleTimeout(int sig)
 	std::cout << "CGIHandler: Timeout" << std::endl;
 }
 
-bool CGIHandler::executeCGI(const MetaVariables &env, HTTPResponse &response)
+bool CGIHandler::executeCGI(const MetaVariables &env, std::string body, HTTPResponse &response)
 {
 	std::cout << RED << "Entering CGIHandler::executeCGI" << RESET << std::endl;
 	std::string cgiOutput;
 	std::vector<std::string> argv = createArgvForExecve(env);
 	std::vector<std::string> envp = env.getForExecve();
 
+	Debug::log("CGIHandler: executeCGI: body: " + body, Debug::NORMAL);
+
 	int pipeFD[2];
+	int bodyPipeFD[2];
```

Comment:
I don't know a lot about the CGI, but I was thinking that the body is getting "into" the CGI process through the STDIN. Why do we need two pipes? The main process write the body in the fd 1 of the pipe, which becomes the STDIN in the child, the CGI read (automatically) form STDIN. And output to STDOUT which is set as the read end of the same pipe from which the body went in. No?

Reply:
Discussed this with the friend. You are right, we need two pipes. Sockets are bi-directional, Pipes are uni-directional.